src/services/ui/src/components/RHF/TableGroup.tsx

```diff
@@ -0,0 +1,173 @@
+import { TableGroupProps } from "shared-types";
+import { Button, FormField } from "../Inputs";
+import {
+  Table,
+  TableBody,
+  TableHead,
+  TableHeader,
+  TableRow,
+  TableCell,
+} from "../Table";
+import { slotInitializer } from "./utils";
+import { RHFSlot } from "./Slot";
+import { cn } from "@/lib";
+
+const FieldRow = <TFields extends FieldValues>(
+  props: { onTrashClick: () => void } & TableGroupProps<TFields>
+) => {
+  return (
+    <TableRow>
+      {props.fields.map((SLOT, i) => {
+        const name = props.name + SLOT.name;
+        return (
+          <TableCell
+            key={`${props.name}cell.${i}`}
+            className={SLOT.tbColumnStyle}
+          >
+            <FormField
+              key={name}
+              control={props.control}
+              name={name as never}
+              {...(SLOT.rules && { rules: SLOT.rules })}
+              render={RHFSlot({
+                ...SLOT,
+                control: props.control,
+                name: name,
+                removeFormDecoration: true,
+                formItemStyling: "gap-0 py-0",
+                props: {
+                  ...SLOT.props,
+                  className: cn(
+                    "border-slate-300",
+                    (SLOT?.props as any)?.classname
+                  ),
+                } as any,
+              })}
+            />
+          </TableCell>
+        );
+      })}
+      {props.scalable && (
+        <TableCell className="p-0 pr-2 py-4 w-14">
+          <Button
+            onClick={props.onTrashClick}
+            variant={"ghost"}
+            className="hover:text-destructive text-primary"
+          >
+            <Trash2 />
+          </Button>
+        </TableCell>
+      )}
+    </TableRow>
+  );
+};
+
+const TableError = (props: { fieldName: string; row: number }) => {
+  const { getFieldState, formState } = useFormContext();
+  const { error } = getFieldState(props.fieldName, formState);
+  const body =
+    error && !Array.isArray(error)
+      ? String(error?.message) + ` - Row ${props.row}`
+      : undefined;
+
+  if (!body) return null;
+
+  return (
+    <p
+      id={`${props.fieldName}-form-item-message`}
+      key={`${props.fieldName}-form-item-message`}
+      className={"text-[0.8rem] font-medium text-destructive"}
+    >
+      {body}
+    </p>
+  );
+};
+
+export const TableGroup = <TFields extends FieldValues>({
+  initNumRows = 1,
+  scalable = true,
+  tableStyle,
+  ...props
+}: TableGroupProps<TFields>) => {
+  const fieldArr = useFieldArray({
+    control: props.control,
+    name: props.name,
+    shouldUnregister: true,
+  });
+
+  const onAppend = () => {
+    fieldArr.append(props.fields.reduce(slotInitializer, {}) as never);
+    console.log("fieldArr", fieldArr);
+  };
+
+  useEffect(() => {
+    if (fieldArr.fields.length) return;
+
+    const initArray = [];
+    do {
+      initArray.push(props.fields.reduce(slotInitializer, {}));
+    } while (initArray.length < initNumRows);
```

Comment:
Cool Solution - threw me off a bit.  Here's how I'd implement it more old-school:

```js
const initArray = [...Array(initNumRows)].map(() =>
  props.fields.reduce(slotInitializer, {})
);
```